### DISCUSSION TITLE
hubserving部署后，服务端提示：This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.

### DISCUSSION THREAD

**Discussion by greatliu** [PaddlePaddle/PaddleOCR#13424] [Q&A]
我使用paddlehub部署的表格识别服务：
(paddlehub) great61@great61-LEGION-REN7000P-26AMR:~/Workspace/PaddleOCR-2.8.0$ export CUDA_VISIBLE_DEVICES=0
(paddlehub) great61@great61-LEGION-REN7000P-26AMR:~/Workspace/PaddleOCR-2.8.0$ hub serving start -c ./deploy/hubserving/structure_table/config.json
/home/great61/miniconda3/envs/paddlehub/lib/python3.9/site-packages/_distutils_hack/__init__.py:26: UserWarning: Setuptools is replacing distutils.
  warnings.warn("Setuptools is replacing distutils.")
[2024-07-18 09:03:22,140] [ WARNING] - The _initialize method in HubModule will soon be deprecated, you can use the __init__() to handle the initialization of the object
use gpu:  True
CUDA_VISIBLE_DEVICES:  0
 * Serving Flask app 'paddlehub.serving.app_compat'
 * Debug mode: off
[2024-07-18 09:03:24,429] [    INFO] _internal.py:97 - WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.
 * Running on all addresses (0.0.0.0)
 * Running on http://127.0.0.1:8869
 * Running on http://192.168.1.2:8869
[2024-07-18 09:03:24,429] [    INFO] _internal.py:97 - Press CTRL+C to quit

如果我是做为生产环境使用，应该更改哪里才能使用wsgi server？

**Comment by GreatV** [comment]
为了在生产环境中使用 PaddleHub 部署的表格识别服务，你需要使用一个生产级别的 WSGI 服务器，而不是默认的 Flask 开发服务器。常见的 WSGI 服务器包括 Gunicorn、uWSGI 等。

**Reply by greatliu** [reply]
我搜了下，其他情况下碰到这个问题，都是说要改几行代码。
那paddle hubserving的demo里这个问题，是不是也要改代码呢？我找不到对应代码所在的文件……

**Reply by SWHL** [reply]
paddle hubserving 应该不维护了，找找其他的平替吧

**Reply by hooyia123** [reply]
遇到了同样的问题，求解决办法或解决方向

**Reply by greatliu** [reply]
> paddle hubserving 应该不维护了，找找其他的平替吧

刚开始我试过paddle serving，里面转换模型代码错误，回复说不维护了。
这次hubserving也说不维护了。

那么，怎么部署paddle服务才是最优解？转换为onnx，用onnx runtime部署，还是自己写flask部署呢？

**Reply by SWHL** [reply]
用onnxruntime或者Flask都可以，具体要都试试。我们这里也没有成熟的部署经验

**Reply by greatliu** [reply]
> 用onnxruntime或者Flask都可以，具体要都试试。我们这里也没有成熟的部署经验

部署这么重要的事情，官方也没给个持续维护的方案。唉。

**Reply by SWHL** [reply]
官方基本都是搞算法的，欢迎贡献哈。大家都是为爱发电。

**Reply by greatliu** [reply]
> 官方基本都是搞算法的，欢迎贡献哈。大家都是为爱发电。

感谢开源作者。我要是能搞就搞了，我也不是专业编程的啊，所以为难。。。
